### PR TITLE
Adding amazoncorretto 17 runtime for java

### PR DESCRIPTION
**Pull request by JohnPreston**
*Issue #, if available:*

*Description of changes:*

Adding amazon corretto 17 to the runtimes.

Tested with

```bash
docker run -it -v /var/run/docker.sock:/var/run/docker.sock -e "IMAGE_NAME=aws/codebuild/standard:5.0.1" -e "ARTIFACTS=/home/john/artifacts/" -e "SOURCE=/home/john/GIT/pycharm/aws-codebuild-docker-images/ubuntu/standard/5.0" -e "BUILDSPEC=/home/john/GIT/pycharm/aws-codebuild-docker-images/ubuntu/standard/5.0/test.yaml" -e "INITIATOR=john" public.ecr.aws/codebuild/local-builds:latest
```

And file for test

```yaml
version: 0.2

phases:
  install:
    runtime-versions:
      java: corretto17
  build:
    commands:
      - java -version

```

And running mvn with sample application

```yaml
version: 0.2

phases:
  install:
    runtime-versions:
      java: corretto17
  build:
    commands:
      - java -version
      - mvn clean package
```

with both al2 and ubuntu, all pass as per my build tests.

### CODE REVIEW

**Comment by JohnPreston** [comment]
@leoherran-aws @alemagna-aws 
Sorry to bother you guys, just unsure what the process is to get this reviewed ?
Thank you :)

File: al2/x86_64/standard/3.0/Dockerfile [hunk 1]
```diff
@@ -226,6 +229,26 @@ RUN set -x \
...
+    # Note: We will use update-alternatives to make sure JDK11 has higher priority for all the tools
+    && export GNUPGHOME="$(mktemp -d)" \
+    && curl -fL -o corretto.key https://yum.corretto.aws/corretto.key \
+    && gpg --batch --import corretto.key \
+    && gpg --batch --export --armor '6DC3636DAE534049C8B94623A122542AB04F24E3' > corretto.key \
+    && rpm --import corretto.key \
+    && rm -r "$GNUPGHOME" corretto.key \
+    && curl -fL -o /etc/yum.repos.d/corretto.repo https://yum.corretto.aws/corretto.repo \
+    && grep -q '^gpgcheck=1' /etc/yum.repos.d/corretto.repo \
+    && yum install -y java-17-amazon-corretto-devel-17.0.0.35-1 \
```
Comment (joschi) [inline al2/x86_64/standard/3.0/Dockerfile:249]: While I think pinning versions in Docker images is useful and should be encouraged, the other versions of Amazon Corretto are not pinned (lines 208 & 231).

Should we "unpin" Amazon Corretto 17 here?
```suggestion
    && yum install -y java-17-amazon-corretto-devel \
```

**Comment by DaanVleugels** [comment]
Is it working with Gradle as well?

**Comment by JohnPreston** [comment]
> Is it working with Gradle as well? I thought Java 17 needed a Gradle version bump to 7.3 (first version fully supporting java 17)

Do you have a sample test that I can use to test this out ? Most stuff I have tested things with was mvn based :/

File: al2/x86_64/standard/3.0/Dockerfile (hunk 1 above)
Comment (JohnPreston) [reply al2/x86_64/standard/3.0/Dockerfile]: Thanks for that. I will check for other potential occurrences of that.

**Comment by ceardigo** [comment]
> Do you have a sample test that I can use to test this out ? Most stuff I have tested things with was mvn based :/

Hello John,

do you mean a sample application using java 17 and gradle? I can provide that if you want.

Thanks

**Comment by JohnPreston** [comment]
> > Do you have a sample test that I can use to test this out ? Most stuff I have tested things with was mvn based :/
> 
> Hello John,
> 
> do you mean a sample application using java 17 and gradle? I can provide that if you want.
> 
> Thanks

Found an open source project I was interested in that uses gradle so will test that :)

**Review by kizdebski — APPROVED** [review]

**Review by ecs-jnguyen — APPROVED** [review]

**Comment by JohnPreston** [comment]
@ceardigo FYI I did test since then (sorry I forgot to update this) and `gradle` tests worked fine.

**Comment by akefirad** [comment]
Is there any update about this PR? Thanks.

**Comment by rasensio** [comment]
Any update for this PR ?

**Comment by JohnPreston** [comment]
@ecs-jnguyen @kizdebski 
Any update on merging this ? Thanks

**Review by lmadhira — APPROVED** [review]

**Comment by JohnPreston** [comment]
Any luck yet on getting thig through ? It's been a few months now :/

**Comment by ym-herby** [comment]
Hi it would be really helpful to get this approved and deployed. Any updates and timeline would be helpful

**Comment by leoherran-aws** [comment]
We're actively working on the release of this change. Will update when it's fully released.

**Review by sl800ad — APPROVED** [review]

**Comment by grisotto** [comment]
Any update on this PR?

**Comment by JohnPreston** [comment]
@Dylan-AWS any reason this was not part of the release 8h ago !?

**Comment by JohnPreston** [comment]
Rebased from master for convenience - just in case.
But when could this be released ?

**Comment by mikelhamer** [comment]
plz

**Comment by subinataws** [comment]
Corretto 17 is supported now with https://github.com/aws/aws-codebuild-docker-images/releases/tag/22.06.30

**Pull request closed without merging**